hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/RowSource.java

```diff
@@ -46,7 +50,9 @@ protected final InputBatch<Dataset<Row>> fetchNewData(Option<String> lastCkptStr
       Dataset<Row> sanitizedRows = SanitizationUtils.sanitizeColumnNamesForAvro(dsr, props);
       SchemaProvider rowSchemaProvider =
           UtilHelpers.createRowBasedSchemaProvider(sanitizedRows.schema(), props, sparkContext);
-      return new InputBatch<>(Option.of(sanitizedRows), res.getValue(), rowSchemaProvider);
+      Dataset<Row> wrappedDf = HoodieSparkUtils.maybeWrapDataFrameWithException(sanitizedRows, HoodieReadFromSourceException.class.getName(),
+          "Failed to read from row source", props.getBoolean(EXTRA_ROW_SOURCE_EXCEPTIONS.key(), EXTRA_ROW_SOURCE_EXCEPTIONS.defaultValue()));
```

Comment:
Use `ConfigUtils#getBooleanWithAltKeys` instead of directly referencing the config key.